### DISCUSSION TITLE
Typo in 6. Viewsets and routers of the tutorial in the docs causing Runtime Error

### DISCUSSION THREAD

**Discussion by BOVAGE** [encode/django-rest-framework#8481] [Potential Issue] [answered]
The **basename** specified in [Using Routers](https://www.django-rest-framework.org/tutorial/6-viewsets-and-routers/#using-routers) should be `snippet` and `user` respectively instead of `snippets` and `users` as stated in the docs. 

### Further details:
Following the docs this way will lead to a runtime error of improperly configured at /snippets/ when accessing the snippet list and detail likewise the user list and detail. 
#### Why?
The **basename** isn't in sync with what has been used earlier in [Hyperlinking Our Api](https://www.django-rest-framework.org/tutorial/5-relationships-and-hyperlinked-apis/#hyperlinking-our-api) of Chap 5. Relationships and Hyperlinking of the tutorial.

I'll love to create an issue and pull request to fix the typo. Thanks!🙏

**Comment by johnt1838** [comment, marked as answer]
ERROR: Could not resolve URL for hyperlinked relationship using view name "snippet-detail". You may have failed to include the related model in your API, or incorrectly configured the `lookup_field` attribute on this field.

**Reply by BOVAGE** [reply]
Have fixed the issue. 

There is a typo in the tutorial section of the documentation. Made a discussion to get a 'go-head' to create an issue.

**Reply by EteimZ** [reply]
@BOVAGE I noticed this too while going through the tutorial.

**Reply by KRicky13** [reply]
@BOVAGE Ran into this issue as well today. Have you been given a green light to fix this yet?

**Reply by lovelydinosaur** [reply]
Ah right, looks like it must have become broken since... https://github.com/encode/django-rest-framework/pull/8349

Which *was* required since newer versions needed an explicit basename, but didn't get the names quite right.

Yup a fix here would be welcome.

**Reply by BOVAGE** [reply]
The documentation explains how the basename is used in generating the url name quite well [here](https://www.django-rest-framework.org/api-guide/routers/#api-guide)

This might help someone.

**Reply by BOVAGE** [reply]
@tomchristie This pull request #8493  fixes the issue.

**Reply by RaSah673** [reply]
For anyone else coming across this question after following the the tutorial, here is the answer.

Change these lines in the snippets/urls.py:

router.register(r'snippets', views.SnippetViewSet,basename="snippets")
router.register(r'users', views.UserViewSet,basename="users")
To these (note the singular basenames)

router.register(r'snippets', views.SnippetViewSet,basename="snippet")
router.register(r'users', views.UserViewSet,basename="user")
